handbook/engineering/commit_messages.md

```diff
@@ -0,0 +1,37 @@
+# Commit message guidelines
+
+Having consistent commit messages makes it easier to see at a glance what code has been affected by recent changes. It also makes it easier to search through our commit history.
+
+The body should contain
+
+To get some background, please read [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/)
+
+## General guidelines
+
+1. Separate subject from body with a blank line
+2. Limit the subject line to 50 characters
```

Comment:
I struggle deeply with 50 characters title width. It's just very hard to do that when the names of things your changing are long, which occurs quite frequently in our codebase.

Reply:
Shorter is better, but I view this as a soft limit in any case. If I write what I think is a good enough commit subject and it happens to be more than 50 characters, I have no problem committing that.

This document uses the word "guidelines" which makes it ambiguous to me if these are "required" or "recommended". I wouldn't be comfortable with this one being "required" but am fine with it being "recommended".

Reply:
My intention was for these to be recommendations, not requirements